web_monitoring/diffing_server.py

```diff
@@ -67,17 +69,47 @@ def get(self, differ):
         try:
             func = self.differs[differ]
         except KeyError:
-            self.send_error(404)
+            self.send_error(404, reason="Diffing method not found.")
             return
 
         # If params repeat, take last one. Decode bytes into unicode strings.
         query_params = {k: v[-1].decode() for k, v in
                         self.request.arguments.items()}
-        a = query_params.pop('a')
-        b = query_params.pop('b')
-
+        try:
+            a = query_params.pop('a')
+            b = query_params.pop('b')
+        except KeyError:
+            self.send_error(
+                        400, reason="Malformed request. A URL parameter is missing.")
```

Comment:
As above, it’s *really* helpful to be able to say *what* parameter was missing. So maybe something like:

```py
self.send_error(
    400,
    reason='Malformed request. You must provide a URL as the value for both `a` and `b` query parameters.')
```

Even better to be able say specifically whether it was `a`, `b`, or both that were missing, but that’s a lot more code and I think the above is probably good enough :)